src/value/tape/cmp.rs

```diff
@@ -9,7 +9,7 @@ impl<'tape, 'input> PartialEq for Value<'tape, 'input> {
     #[cfg_attr(not(feature = "no-inline"), inline)]
     #[must_use]
     fn eq(&self, other: &Self) -> bool {
-        self == other
+        self.0 == other.0
```

Comment:
I believe the previous implementation was buggy, the clippy lint indicates that it would have looped infinitely.

I think comparing the wrapped value is what was intended here?

Reply:
Yes that's right! Good catch :)

I stumbled about that the other day too, clippy is my secret hero to be honest 😂 it avoids so many issues.